### ISSUE TITLE
Add option to exclude Dark Pool Data from Aggregate Bars

### ISSUE THREAD

**Issue by ParsaPilehvari** [polygon-io/feature-requests#39]
The aggregate bar endpoints have the dark pool trades and volume rolled into it, leading to moments in my datasets where the high, low, and close of a bar can be wildly different from what the NASDAQ historical data reports in reality. It seems counter-intuitive to include this data in the aggregate bars as retail traders are unable to actually trade at those dark pool values, which only results in noise to an algorithm and with sufficient dark pool volumes, it can easily trigger false positives/negatives.

Having a boolean option to filter out the dark pool exchange's OHLCV from the aggregate bars will resolve this issue.

Note: A potential workaround could be consolidating the historical trades and filtering out the FINRA dark pool exchange directly, but during high volume periods, this will require multiple API calls to consolidate each trade into an aggregate bar, which severely degrades real-time performance.

**Comment by ParsaPilehvari** [comment]
Seems to be a duplicate of [Add option to exclude dark pool from aggregate data](https://github.com/polygon-io/feature-requests/issues/1) however it does not seem like the issue was actually resolved.

**Comment by swilson2020** [comment]
Yes, this is a common complaint.  I've also been requesting this.  I have to pull clean data from Schwab for trading and backtesting.

**Comment by ParsaPilehvari** [comment]
I am still encountering this issue and the 8am untradeable spikes are skewing the results for my live and backtest algorithms. I like financial modeling prep a lot, but I find they update their bars too slowly for my liking. It only really impacts the speed of my live algos.

**Comment by swilson2020** [comment]
I'm finding volume errors on FMP and I'm working with them on it.  Basically their 9:30AM candles are reporting volume that none of the other providers do.  Sometimes to the tune of 12,000% more.  

So yeah, I might have to build my own aggregates from Polygon's data.  Which is dumb, means more bandwidth and more API usage simply because they won't offer clean aggregates.

**Comment by ParsaPilehvari** [comment]
I remember facing the same issue, I assumed it was because of what exchanges it had access to. If you're in one of the supported countries, Alpaca might be a better data provider, considering they are cheaper although they also have the same dark pool issue.

They don't offer bars less than one minute but the APIs are pretty nice. I think both them and Polygon use similar data sources for their data.

**Comment by gearoid-murphy** [comment]
I work around this by downloading the trades and excluding any trades with the trf_id field set. It would be much more convenient if we could have these excluded when requesting aggregate bars (candles).
